pkg/plugins/logger.go

```diff
@@ -0,0 +1,32 @@
+package plugins
+
+import (
+	"context"
+
+	"github.com/http-wasm/http-wasm-host-go/api"
+	"github.com/rs/zerolog"
+)
+
+func initWasmLogger(logger *zerolog.Logger) *WasmLogger {
+	return &WasmLogger{
+		logger: logger,
+	}
+}
+
+// compile-time check to ensure ConsoleLogger implements api.Logger.
+var _ api.Logger = WasmLogger{}
+
+// WasmLogger is a convenience which writes anything above LogLevelInfo to os.Stdout.
+type WasmLogger struct {
+	logger *zerolog.Logger
+}
+
+// IsEnabled implements the same method as documented on api.Logger.
+func (w WasmLogger) IsEnabled(level api.LogLevel) bool {
+	return true
+}
+
+// Log implements the same method as documented on api.Logger.
+func (w WasmLogger) Log(_ context.Context, level api.LogLevel, message string) {
+	w.logger.WithLevel(zerolog.Level(level + 1)).Msg(message)
```

Comment:
Shall we do `w.logger.WithContext(ctx).WithLevel(zerolog.Level(level + 1)).Msg(message)` here?

Reply:
<img width="1277" alt="Screenshot 2023-10-25 at 10 43 39" src="https://github.com/traefik/traefik/assets/22482503/49130ce3-506a-4315-8822-000c776fc461">

zerolog.UpdateContext is perhaps the correct one instead. However, not sure should this be really implemented

Reply:
My bad. I meant `logger.Ctx(ctx)`

Reply:
it does not have that either :)